aravis/src/manual/buffer.rs

```diff
@@ -13,6 +14,27 @@ pub enum ImageError {
 }
 
 impl Buffer {
+
+	/// Create an Aravis buffer that owns its own data from a pre-allocated raw buffer.
+	///
+	/// The `destroy_callback` argument is called to destroy the buffer, and should free the resources associated with the buffer.
+	///
+	/// # Safety
+	/// The pointer and length parameter must indicate a valid memory region where Aravis can safely write data to until the `destroy_callback` is called.
+	pub fn new_preallocated_owned<F: FnOnce()>(data: *mut u8, len: usize, destroy_callback: F) -> Self {
```

Comment:
Shall we be nice and pass the data pointer to the callback?

Reply:
Seems kind of confusing for the user that's writing the callback.  My thinking was that they should just move everything they need for the desctructor into a lambda and be done with it.

Maybe I am misunderstanding your suggestion though.

Reply:
Are you suggesting that they write a callback such that `F: FnOnce(*mut u8)`?

Reply:
Yeah. You're right that it is hopefully normally not needed. In that sense I can see the logic in not providing it by default.

Fair enough, lets leave it out :)

Reply:
It not really more flexible, just simpler if you actually need the data pointer. But you rarely need it.

If you really need it, you can capture a copy of it for the callback.